diff --git a/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr-common.dtsi b/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr-common.dtsi
index 604b329d61..2871f8b21b 100644
--- a/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr-common.dtsi
+++ b/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr-common.dtsi
@@ -213,14 +213,14 @@
 
 			band@0 {
 				reg = <0>;
-				nvmem-cells = <&macaddr_factory_1e001c 0>;
+				nvmem-cells = <&macaddr_factory_4 0>;
 				nvmem-cell-names = "mac-address";
 			};
 
 			/* FIXME: this does not work */
 			band@1 {
 				reg = <1>;
-				nvmem-cells = <&macaddr_factory_1e001c 2>;
+				nvmem-cells = <&macaddr_factory_a 0>;
 				nvmem-cell-names = "mac-address";
 			};
 		};
@@ -287,6 +287,18 @@
 						reg = <0x0 0x1e00>;
 					};
 
+					macaddr_factory_4: eeprom@4 {
+						compatible = "mac-base";
+						reg = <0x4 0x6>;
+						#nvmem-cell-cells = <1>;
+					};
+
+					macaddr_factory_a: eeprom@a {
+						compatible = "mac-base";
+						reg = <0xa 0x6>;
+						#nvmem-cell-cells = <1>;
+					};
+
 					macaddr_factory_1e001c: eeprom@1e001c {
 						compatible = "mac-base";
 						reg = <0x1e001c 0x6>;
diff --git a/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr7250-v1.dts b/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr7250-v1.dts
index 5020505e7d..5f6ccff87b 100644
--- a/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr7250-v1.dts
+++ b/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr7250-v1.dts
@@ -7,14 +7,14 @@
 #include "mt7988d-tplink-tl-7dr-common.dtsi"
 
 / {
-	model = "TP-Link TL-7DR7250 v1";
+	model = "TP-Link TL-7DR7230 Rev1.0 SP2";
 	compatible = "tplink,tl-7dr7250-v1", "mediatek,mt7988d";
 };
 
 &gmac1 {
 	nvmem-cells = <&macaddr_factory_1e001c 0>;
 	nvmem-cell-names = "mac-address";
-	openwrt,netdev-name = "lan3";
+	label = "lan3";
 };
 
 &gmac2 {
@@ -22,7 +22,7 @@
 	phy = <&phy13>;
 	nvmem-cells = <&macaddr_factory_1e001c 1>;
 	nvmem-cell-names = "mac-address";
-	openwrt,netdev-name = "wan";
+	label = "wan";
 	status = "okay";
 };
 
@@ -35,11 +35,11 @@
 };
 
 &gsw_port0 {
-	label = "lan2";
+	label = "lan1";
 };
 
 &gsw_port1 {
-	label = "lan1";
+	label = "lan2";
 };
 
 &gsw_port2 {
diff --git a/target/linux/mediatek/filogic/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac b/target/linux/mediatek/filogic/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac
index 4113811fd1..c70ac75837 100644
--- a/target/linux/mediatek/filogic/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac
+++ b/target/linux/mediatek/filogic/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac
@@ -173,6 +173,9 @@ case "$board" in
 	tplink,tl-xtr8488)
 		[ "$PHYNBR" = "1" ] && get_mac_label > /sys${DEVPATH}/macaddress
 		;;
+	tplink,tl-7dr7250-v1)
+		[ "$PHYNBR" = "1" ] && mtd_get_mac_binary factory 0xa > /sys${DEVPATH}/macaddress
+		;;
 	ubnt,unifi-6-plus)
 		addr=$(mtd_get_mac_binary EEPROM 0x6)
 		[ "$PHYNBR" = "0" ] && echo "$addr" > /sys${DEVPATH}/macaddress
