diff --git a/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr-common.dtsi b/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr-common.dtsi
index 604b329d61..17e9f60fdb 100644
--- a/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr-common.dtsi
+++ b/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr-common.dtsi
@@ -93,7 +93,7 @@
 };
 
 &gmac0 {
-	nvmem-cells = <&macaddr_factory_1e001c 0>;
+	nvmem-cells = <&macaddr_factory_36001c 0>;
 	nvmem-cell-names = "mac-address";
 	status = "okay";
 };
@@ -213,14 +213,14 @@
 
 			band@0 {
 				reg = <0>;
-				nvmem-cells = <&macaddr_factory_1e001c 0>;
+				nvmem-cells = <&macaddr_factory_4 0>;
 				nvmem-cell-names = "mac-address";
 			};
 
 			/* FIXME: this does not work */
 			band@1 {
 				reg = <1>;
-				nvmem-cells = <&macaddr_factory_1e001c 2>;
+				nvmem-cells = <&macaddr_factory_a 0>;
 				nvmem-cell-names = "mac-address";
 			};
 		};
@@ -287,9 +287,21 @@
 						reg = <0x0 0x1e00>;
 					};
 
-					macaddr_factory_1e001c: eeprom@1e001c {
+					macaddr_factory_4: eeprom@4 {
 						compatible = "mac-base";
-						reg = <0x1e001c 0x6>;
+						reg = <0x4 0x6>;
+						#nvmem-cell-cells = <1>;
+					};
+
+					macaddr_factory_a: eeprom@a {
+						compatible = "mac-base";
+						reg = <0xa 0x6>;
+						#nvmem-cell-cells = <1>;
+					};
+
+					macaddr_factory_36001c: eeprom@36001c {
+						compatible = "mac-base";
+						reg = <0x36001c 0x6>;
 						#nvmem-cell-cells = <1>;
 					};
 				};
diff --git a/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr7250-v1.dts b/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr7250-v1.dts
index 5020505e7d..f15461d602 100644
--- a/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr7250-v1.dts
+++ b/target/linux/mediatek/dts/mt7988d-tplink-tl-7dr7250-v1.dts
@@ -7,12 +7,12 @@
 #include "mt7988d-tplink-tl-7dr-common.dtsi"
 
 / {
-	model = "TP-Link TL-7DR7250 v1";
+	model = "TP-Link TL-7DR7230 Rev1.0 SP2";
 	compatible = "tplink,tl-7dr7250-v1", "mediatek,mt7988d";
 };
 
 &gmac1 {
-	nvmem-cells = <&macaddr_factory_1e001c 0>;
+	nvmem-cells = <&macaddr_factory_36001c 0>;
 	nvmem-cell-names = "mac-address";
 	openwrt,netdev-name = "lan3";
 };
@@ -20,7 +20,7 @@
 &gmac2 {
 	phy-mode = "2500base-x";
 	phy = <&phy13>;
-	nvmem-cells = <&macaddr_factory_1e001c 1>;
+	nvmem-cells = <&macaddr_factory_36001c 1>;
 	nvmem-cell-names = "mac-address";
 	openwrt,netdev-name = "wan";
 	status = "okay";
